dashboard/app/views/devise/registrations/edit.html.haml

```diff
@@ -23,9 +25,12 @@
     - else
       .field
         = f.label :email
-        - if f.object.can_edit_email?
+        - if current_user.can_edit_email? && !current_user.oauth?
```

Comment:
Oauth users technically have to be able to edit email address in order to upgrade from student to teacher and back, but we don't want to have a field for it on the form, so we show the non-editable version instead, and fill in the hidden field from the modal.